server/core/lib/runners/job-handlers/video-storyboard-job-handler.ts

```diff
@@ -0,0 +1,167 @@
+  async create (options: CreateOptions) {
+    const { videoUUID, priority, federateAfter } = options
+
+    const jobUUID = buildUUID()
+
+    const video = await VideoModel.loadFull(videoUUID)
+    // Compute sprites metadata on server side for consistency
+    const inputFile = video.getMaxQualityFile(VideoFileStream.VIDEO)
+    let spriteHeight: number
+    let spriteWidth: number
+    let duration = 0
+
+    await VideoPathManager.Instance.makeAvailableVideoFile(inputFile, async videoPath => {
+      const probe = await ffprobePromise(videoPath)
+      const videoStreamInfo = await getVideoStreamDimensionsInfo(videoPath, probe)
+
+      duration = video.duration
+
+      if (videoStreamInfo.isPortraitMode) {
+        spriteHeight = STORYBOARD.SPRITE_MAX_SIZE
+        spriteWidth = Math.round(spriteHeight * videoStreamInfo.ratio)
+      } else {
+        spriteWidth = STORYBOARD.SPRITE_MAX_SIZE
+        spriteHeight = Math.round(spriteWidth / videoStreamInfo.ratio)
+      }
+    })
+
+    const { spriteDuration, totalSprites } = buildSpritesMetadata({ duration })
+    const spritesCount = findGridSize({ toFind: totalSprites, maxEdgeCount: STORYBOARD.SPRITES_MAX_EDGE_COUNT })
+
+    const payload: RunnerJobGenerateStoryboardPayload = {
+      input: {
+        videoFileUrl: generateRunnerTranscodingVideoInputFileUrl(jobUUID, videoUUID)
+      },
+      sprites: {
+        size: { height: spriteHeight, width: spriteWidth },
+        count: spritesCount,
+        duration: spriteDuration
+      },
+      output: {}
+    }
+
+    const privatePayload: RunnerJobGenerateStoryboardPrivatePayload = {
+      videoUUID,
+      federate: !!federateAfter
+    }
+
+    const job = await this.createRunnerJob({
+      type: 'generate-video-storyboard',
+      jobUUID,
+      payload,
+      privatePayload,
+      priority
+    })
+
+    return job
+  }
+
+  protected isAbortSupported () {
+    return true
+  }
+
+  protected specificUpdate (_options: { runnerJob: MRunnerJob }) {}
+  protected specificAbort (_options: { runnerJob: MRunnerJob }) {}
+
+  // When runner returns the storyboard image, finish the server-side creation like local job would
+  protected async specificComplete (options: { runnerJob: MRunnerJob, resultPayload: GenerateStoryboardSuccess }) {
+    const { runnerJob, resultPayload } = options
+
+    const video = await VideoModel.loadFull(runnerJob.privatePayload.videoUUID)
+    if (!video) return
+
+    const destinationFilename = generateImageFilename()
+    const destinationPath = join(CONFIG.STORAGE.STORYBOARDS_DIR, destinationFilename)
+
+    await move(resultPayload.storyboardFile as string, destinationPath)
+
+    const imageSize = await getImageSizeFromWorker(destinationPath)
+
+    const sprites = (runnerJob.payload as RunnerJobGenerateStoryboardPayload).sprites
+
+    await retryTransactionWrapper(() => {
```

Comment:
Can you also refactor this part to share code with `server/core/lib/job-queue/handlers/generate-storyboard.ts`?